server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServerResourceManager.java

```diff
@@ -178,14 +180,17 @@ private ExecutorService createPriorityExecutor(ScanExecutorConfig sec,
         Comparator<ScanInfo> comparator =
             factory.createComparator(new ScanPrioritizer.CreateParameters() {
 
+              private final Supplier<ServiceEnvironment> senvSupplier =
+                  Suppliers.memoize(() -> new ServiceEnvironmentImpl(context));
```

Comment:
It's not obvious here why this instance needs to be lazily loaded when other instances are not.

Reply:
I reviewed that further and found it does not need to be lazily loaded.  Its only created once when a scan thread pool is created, so there is no per scan performance concern.

Reply:
Oh removed the memoize() call.